Composer/packages/client/src/shell/triggerApi.ts

```diff
@@ -28,7 +28,7 @@ function createTriggerApi(
     return value.substring(startIndex + 1, endIndex);
   };
 
-  const createTriggerHandler = async (id, formData, url) => {
+  const createTriggerHandler = async (id, formData, url, checkDuplicatedKind) => {
```

Comment:
Don't fix the issue by changing the interface of `createTriggerHandler`, it shoudn't be affected by this issue.
Hoist `checkDuplicatedKind` logic to invoker (DesignPage in your case).